Branch-SDK/src/io/branch/referral/Branch.java

```diff
@@ -2332,14 +2334,15 @@ public void applyReferralCode(final String code, final BranchReferralInitListene
      * user selected clients.</p>
      *
      * @param builder A {@link io.branch.referral.Branch.ShareLinkBuilder} instance to build share link.
+     * @return The instance of the {@link Dialog} holding the share view.
      */
-    private void shareLink(ShareLinkBuilder builder) {
+    private Dialog shareLink(ShareLinkBuilder builder) {
```

Comment:
@shortstuffsushi  Another thinking on this is to keep the `shareLink()` method as it is now to return void. Then add a new method `createShareLinkDialog()` method to get the instance of the dialog is they want to Handle it by them self. How do you think about that?